assets/scripts/menus/MenuBar.jsx

```diff
@@ -0,0 +1,170 @@
+    const signInVisibilityStyle = userId
+      ? { display: 'none' } : {}
+
+    // Buttons have `disabled={false}` because
+    // Firefox sometimes disables some buttons… unsure why
+    // --
+    // Note on translations - the {t()} method doesn't re-render when the
+    // language changes, so this still uses the `data-i18n` method to pick up
+    // on text content changes. Individual menus will re-render with {t()}.
+    return (
+      <nav className='menu-bar'>
+        <ul className='menu-bar-left'>
+          <li className='menu-bar-title'>
+            <div className='streetmix-logo' />
+            <h1>Streetmix</h1>
+          </li>
+          <li id='help-menu-item'>
+            <button
+              data-name='help'
+              data-i18n='menu.item.help'
+              className='menu-attached'
+              disabled={false}
+              onClick={this.onClickMenuButton}
+            >
+              Help
+            </button>
+          </li>
+          <li id='contact-menu-item'>
+            <button
+              data-name='contact'
+              data-i18n='menu.item.contact'
+              className='menu-attached'
+              disabled={false}
+              onClick={this.onClickMenuButton}
+            >
+              Contact
+            </button>
+          </li>
+        </ul>
+        <ul className='menu-bar-right'>
+          <li id='identity-menu-item' style={identityMenuVisibilityStyle}>
+            <button
+              data-name='identity'
+              className='menu-attached'
+              disabled={false}
+              onClick={this.onClickMenuButton}
+            >
+              <div className='avatar' data-user-id={userId} />
+              <span className='user-id'>{userId}</span>
+            </button>
+          </li>
+          <li id='sign-in-menu-item' style={signInVisibilityStyle}>
+            <a href={`/${URL_SIGN_IN_REDIRECT}`} className='command' data-i18n='menu.item.sign-in'>
+              Sign in
+            </a>
+          </li>
+          <li>
+            <a href='/new' target='_blank' data-i18n='menu.item.new-street'>
+              New street
+            </a>
+          </li>
+          <li id='gallery-link'>
+            <a href={myStreetsLink} data-i18n='menu.item.my-streets' onClick={onMyStreetsClick}>
+              My streets
+            </a>
+          </li>
+          {
+            (() => {
+              if (debug.experimental) {
+                return (
+                  <li id='settings-menu-item'>
+                    <button
+                      data-name='settings'
+                      data-i18n='menu.item.settings'
+                      className='menu-attached'
+                      disabled={false}
+                      onClick={this.onClickMenuButton}
+                    >
+                      Settings
+                    </button>
+                  </li>
+                )
+              }
+            })()
```

Comment:
Wow, that’s a lot of work Lou! I’m not following everything perfectly, but if the no-internet mode of street mix is easy to enable, I’d be curious to hear about it. If it’s difficult and would take a lot of time though, no worries. Would it still log each street design entry and record data as normal (e.g.creating URLs of /167, /168, /169…) even without being connected to the internet?

I’m also working on an info sheet to help guide people that want to do future custom deployments of Streetmix. I’m working off Ezra’s list he gave me (below). If you guys have anything you think would be good to add let me know! I’ll put the Illustrator/Indesign file and PDF on Slack, The Forums, or wherever it makes sense when complete.

-Trey

1. Fork Streetmix into your own repository on GitHub.
2. Clone the Streetmix code onto your computer using Git.
3. Modify Streetmix for your environment and needs (change text, images, URL configuration, etc.).
4. Test the changes locally. (see CONTRIBUTING.md for instructions to set up a development environment).
5. Commit the changes and push to your fork.
6. Create a Heroku app for your deployment. Configure a custom domain name on Heroku if needed.
7. Configure a MongoDB addon on your Heroku app.
8. Push your fork to your Heroku app.
9. Set up other credentials and addons as necessary (i.e. Twitter for logins, Sendgrid for email).
10. If you make additional changes, commit them to Git, push to GitHub, and push to your Heroku app.

> On Sep 6, 2016, at 11:03 AM, Lou Huang <notifications@github.com> wrote:
> 
> In assets/scripts/menus/MenuBar.jsx <https://github.com/codeforamerica/streetmix/pull/582#discussion_r77652973>:
> 
> > +              if (debug.experimental) {
> > +                return (
> > +                  <li id='settings-menu-item'>
> > +                    <button
> > +                      data-name='settings'
> > +                      data-i18n='menu.item.settings'
> > +                      className='menu-attached'
> > +                      disabled={false}
> > +                      onClick={this.onClickMenuButton}
> > +                    >
> > +                      Settings
> > +                    </button>
> > +                  </li>
> > +                )
> > +              }
> > +            })()
> That's a good idea. It may also be better to think of these menu items as their own components? I didn't address it here, but a little-used feature for "no-internet mode" (deploying Streetmix in public environments where you don't have access to Internet) disables a bunch of these buttons as well, but only does so via CSS by hiding them. It may be worth unifying the logic behind conditionally showing menu items...
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/codeforamerica/streetmix/pull/582/files/56682ae23c08450123b8fcb56450cf98d53b69ec#r77652973>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AL-HbHS7_LlgsRV2NqeSshTNMhC0rVUwks5qnYCrgaJpZM4J1Zkn>.
>